src/main/java/net/minestom/server/instance/block/predicate/CollectionPredicate.java

```diff
@@ -0,0 +1,97 @@
+package net.minestom.server.instance.block.predicate;
+
+import net.minestom.server.codec.Codec;
+import net.minestom.server.codec.StructCodec;
+import net.minestom.server.utils.Range;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+public record CollectionPredicate<T, P extends Predicate<T>>(Contains<T, P> contains, Count<T, P> counts,
```

Comment:
Should probably at least put `@Nullable` on the nullable fields (you omit the `@Nullable` in other field definitions too but I'm just mentioning it here), and preferably also `@NotNull` on the non-nullable ones.